File: Exercise/LuoGu/Code_2024_07_22/code01pro.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        size = len(nums)
        for i in range(size):
            for j in range(i+1, size):
                if nums[i] + nums[j] == target:
                    return [i, j]


Solution().twoSum([3, 2, 3], 6)


def twoSum(nums, target):
    hash_table = {}

    for i in range(len(nums)):
        complement = target - nums[i]

        if complement in hash_table:
            return [hash_table[complement], i]

        hash_table[nums[i]] = i  # 将元素和它的索引存入哈希表中

    return []


print(twoSum([3, 3], 6))

File: Exercise/LuoGu/Code_2024_07_22/code01pro.ts
function twoSum(nums: number[], target: number): number[] {
    let size = nums.length;
    let curSum = 0;
    for (let i = 0, j = 0; i < size && j <= size;) {
        if (curSum == target) {
            return [i, j - 1];
        } else if (curSum < target) {
            curSum += nums[j++];
        } else if (curSum > target) {
            curSum -= nums[i++];
        }
    }
    return []
};

console.log(twoSum([2, 7, 11, 15], 9));
console.log(twoSum([3, 2, 4], 6));
console.log(twoSum([3, 3], 6));